src/app/core/services/search-criteria/search-criteria.service.ts

```diff
@@ -163,471 +46,366 @@ export class SearchCriteriaService {
-    await this.updateRouterQueryParams(params);
+  resetSearchCriteria() {
+    this.competence.set(undefined);
+    this.regions.set(undefined);
+    this.regTypes.set(undefined);
+    this.nickName.set(undefined);
+    this._slushFlow.set(false);
+    this.userSettingService.resetDaysBackForCurrentGeoHazard(); // TODO: Test
+    this.useDaysBack.set(true);
   }
 
-  private toUrlParams(criteria: SearchCriteriaRequestDto, daysBack: number | null): UrlParams {
-    const params = new UrlParams();
-    params.set(URL_PARAM_GEOHAZARD, arrayToSeparatedString(criteria.SelectedGeoHazards));
-    if (daysBack != null) {
-      params.set(URL_PARAM_DAYSBACK, daysBack.toString()); // Convert to string so that 0 is accepted as a value
-      params.delete(URL_PARAM_FROMDATE);
-      params.delete(URL_PARAM_TODATE);
-    } else {
-      params.delete(URL_PARAM_DAYSBACK);
-      params.set(URL_PARAM_FROMDATE, isoDateTimeToLocalDate(criteria.FromDtObsTime));
-      params.set(URL_PARAM_TODATE, isoDateTimeToLocalDate(criteria.ToDtObsTime));
-    }
-    params.set(URL_PARAM_NICKNAME, criteria.ObserverNickName);
-    params.set(URL_PARAM_COMPETENCE, competenceFromDtoToUrl(criteria.ObserverCompetence));
-    params.set(URL_PARAM_REGISTRATION_TYPE, convertRegTypeDtoToUrl(criteria.SelectedRegistrationTypes));
-    params.set(URL_PARAM_ORDER_BY, convertApiOrderByToUrl(criteria.OrderBy as SearchCriteriaOrderBy));
-    params.set(URL_PARAM_REGION, arrayToSeparatedString(criteria.SelectedRegions));
-
-    if (this.isSlushFlow(criteria)) {
-      params.set(URL_PARAM_SLUSH_FLOW, true);
-    } else {
-      params.delete(URL_PARAM_SLUSH_FLOW);
-    }
-
-    if (isValidExtent(criteria.Extent)) {
-      params.set(URL_PARAM_NW_LAT, +criteria.Extent.TopLeft.Latitude.toFixed(ULR_COORDS_PRECISION));
-      params.set(URL_PARAM_NW_LON, +criteria.Extent.TopLeft.Longitude.toFixed(ULR_COORDS_PRECISION));
-      params.set(URL_PARAM_SE_LAT, +criteria.Extent.BottomRight.Latitude.toFixed(ULR_COORDS_PRECISION));
-      params.set(URL_PARAM_SE_LON, +criteria.Extent.BottomRight.Longitude.toFixed(ULR_COORDS_PRECISION));
-    } else {
-      params.delete(URL_PARAM_NW_LAT);
-      params.delete(URL_PARAM_NW_LON);
-      params.delete(URL_PARAM_SE_LAT);
-      params.delete(URL_PARAM_SE_LON);
-    }
-    return params;
+  async applyQueryParams(withExtent = true) {
+    const criteria = withExtent ? this.criteriaWithExtent() : this.criteria();
+    const daysBack = this.daysBack();
+    const useDaysBack = this.useDaysBack();
+    await this.queryParams.apply({ criteria, daysBack: useDaysBack ? daysBack : undefined });
   }
 
-  private async updateRouterQueryParams(params: UrlParams) {
-    const queryParams = params.entries();
-    await this.router.navigate([], {
-      relativeTo: this.activatedRoute,
-      queryParams,
-      replaceUrl: true,
+  addRegion(region: Exclude<Flatten<SearchCriteriaRequestDto['SelectedRegions']>, undefined>) {
```

Comment:
Typen til region-parameteren synes jeg er vanskelig å forstå.
Jeg tror jeg skjønner at SearchCriteriaRequestDto['SelectedRegions'] sikrer at typen endrer seg om man endrer type for SelectedRegions i SearchCriteriaRequestDto. Men Exclude<Flatten... gjorde det ikke noe enklere. 
Om det hadde stått region: number[] i stedet tror jeg fordelene med lesbarhet trumfer robusthet. Hadde vært noe annet om man faktisk hadde en sjekk på hvilke regionsnumre som er gyldige, men tror ikke vi har det?
Kommentaren gjelder forsåvidt alle funksjoner med parametre av typen SearchCriteriaRequestDto['<felt>'], men særlig der vi legger på Flatten eller Exclude.